logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/Conversation.kt

```diff
@@ -8,7 +8,7 @@ import com.wire.kalium.logic.data.user.SelfUser
 import com.wire.kalium.logic.data.user.User
 import com.wire.kalium.logic.data.user.UserId
 
-data class Conversation(val id: ConversationId, val name: String?, val type: Type, val teamId: TeamId?) {
+data class Conversation(val id: ConversationId, val name: String?, val type: Type, val teamId: TeamId?, val groupId: String?) {
```

Comment:
The name `groupId` makes me think of `Conversation.Type.GROUP`.

Is it specific to MLS? Could we name it that way?

Will `Conversation` have a `val protocol: ConvProtocol` too?

What about something like:

```kotlin

data class Conversation(
    val id: ConversationId,
    val name: String?,
    val type: Type,
    val teamId: TeamId?,
    val encryptionInfo: ProtocolInfo
)

sealed class ProtocolInfo{ 
    object Proteus: ProtocolInfo()
    data class MLS(val groupId: String, val groupState: GroupState): ProtocolInfo()
}
```

Reply:
I did this at the persistence level and choose to not expose any of this at the logic level for now.